controller/src/main/java/io/pravega/controller/store/stream/records/SubscriberConfiguration.java

```diff
@@ -0,0 +1,90 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.controller.store.stream.records;
+
+import com.google.common.collect.ImmutableMap;
+import io.pravega.common.ObjectBuilder;
+import io.pravega.common.io.serialization.RevisionDataInput;
+import io.pravega.common.io.serialization.RevisionDataOutput;
+import io.pravega.common.io.serialization.VersionedSerializer;
+import lombok.Builder;
+import lombok.Data;
+import lombok.SneakyThrows;
+
+import java.io.DataInput;
+import java.io.DataOutput;
+import java.io.IOException;
+import java.io.Serializable;
+
+/**
+ * The configuration of a Stream Subscriber.
+ */
+@Data
+@Builder
+public class SubscriberConfiguration implements Serializable {
+
+    public static final SubscriberConfigurationSerializer SERIALIZER = new SubscriberConfigurationSerializer();
+    public static final SubscriberConfiguration EMPTY = new SubscriberConfiguration(0L, ImmutableMap.of());
+    private static final long serialVersionUID = 1L;
```

Comment:
why do we need this?

Reply:
'SubscriberConfiguration' encapsulates all and any data that needs to be stored about a Stream Subscriber. Today it has only the truncation StreamCut but could have more stuff in the future.

Reply:
removed now. fixed.